frontends/web/src/new_front/components/Tables/Leaderboard/TaskModelLeaderboardCard.jsx

```diff
@@ -91,65 +100,134 @@ const TaskModelLeaderboardCard = ({
+    setIsLoading(true);
+    const currentArgs = JSON.stringify({
+      task,
+      context,
+      sort,
+      page,
+    });
+
+    if (initialWeights.current === currentArgs) {
       return;
     }
+    initialWeights.current = currentArgs;
+    try {
+      getInitialWeights(task, context.api, (result) => {
+        if (
+          result.orderedMetricWeights?.length &&
+          result.orderedDatasetWeights?.length
+        ) {
+          setMetrics(result.orderedMetricWeights);
+          setDatasetWeights(result.orderedDatasetWeights);
+          setDescription(result.description);
+        } else {
+          console.error("Invalid initial weights format:", result);
+        }
+      });
+    } catch (error) {
+      console.error("Error fetching initial weights:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [context.api, task, getInitialWeights]);
+
+  const weightCalculations = useMemo(() => {
+    if (!datasetWeights?.length || !metrics?.length) {
+      return null;
+    }
+    const datasetWeightsList = datasetWeights.map(
+      (obj) =>
+        obj.weight / datasetWeights.reduce((sum, item) => sum + item.weight, 0)
+    );
 
-    const currentDirection = sort.direction;
+    const metricWeightsList = metrics.map(
+      (obj) => obj.weight / metrics.reduce((sum, item) => sum + item.weight, 0)
+    );
 
-    const newDirection =
-      field !== sort.field
-        ? SortDirection.DESC
-        : SortDirection.getOppositeDirection(currentDirection);
+    return {
+      datasetWeightsList,
+      metricWeightsList,
+      metricsIDs: metrics.map((metric) => metric.id),
+    };
+  }, [datasetWeights, metrics]);
 
-    setSort({
-      field: field,
-      direction: newDirection,
-    });
-  };
+  const handleScoreData = useCallback(
+    async (datasetWeightsList, metricWeightsList, metricIds) => {
+      setIsLoading(true);
+      try {
+        const scoreData = await post("/task/get_dynaboard_info_by_task_id/", {
+          task_id: taskId,
+          ordered_metric_weights: metricWeightsList,
+          ordered_scoring_dataset_weights: datasetWeightsList,
+          sort_by: sort?.field,
+          sort_direction: sort?.direction,
+          offset: page * pageLimit,
+          limit: pageLimit,
+          metrics: metricIds,
+        });
+        if (response.ok) {
+          setData(scoreData.data);
+          setTotal(scoreData.count);
+        }
+      } catch (error) {
+        console.error("Error fetching score data:", error);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [taskId, post, sort?.field, sort?.direction, page, pageLimit, response]
+  );
 
   useEffect(() => {
-    getInitialWeights(task, context.api, (result) => {
-      setMetrics(result.orderedMetricWeights);
-      setDatasetWeights(result.orderedDatasetWeights);
-      setDescription(result.description);
+    if (!weightCalculations) {
+      console.log("Waiting for weight calculations...");
```

Comment:
Is there a reason to keep this one?

Reply:
I've change it to a comment to understand what is happening there